tmt/steps/provision/artemis.py

```diff
@@ -279,6 +279,9 @@ class ProvisionArtemis(tmt.steps.provision.ProvisionPlugin):
     # Guest instance
     _guest = None
 
+    # Provide additional options
+    _guest_class = tmt.steps.provision.GuestSsh
```

Comment:
`_guest_class` should point to `ArtemisGuest`.

This applies to other modules, too: if they have their own guest class, the plugin's `_guest_class` must point to this class.

Reply:
Ok. But I'm not sure how to handle this without moving ArtemisGuest class on top of ProvisionArtemis because there is already:

NameError: name 'GuestArtemis' is not defined

erorr

Reply:
Yes, split into two commits helped a lot.

Reply:
Ok. Thanks for review.

Reply:
NOTE: just before split to more commits.

Reply:
Ok, let me know, when the two commits are ready for review.

Reply:
Thanks. Any ideas how to rebase to change #1654?
So split is now not needed if I understand it correctly.

Reply:
A rebase on top of the branch in #1654 should do the trick.

Reply:
Rebased on top of https://github.com/teemtee/tmt/pull/1654

Reply:
And also modify this MR, to use `reorder-guests-and-plugins` branch as its target, Github should than show just the changes you made.